packages/demo/src/FavoriteColorsForm.re

```diff
@@ -0,0 +1,119 @@
+module FavoriteColorsLenses = [%lenses
+  type t = {
+    name: string,
+    hex: string,
+  }
+];
+
+module StateLenses = [%lenses
+  type state = {
+    name: string,
+    favoriteColors: array(FavoriteColorsLenses.t),
+  }
+];
+
+module Form = ReFormNext.Make(StateLenses);
+
+[@react.component]
+let make = () => {
+  let {
+    state,
+    submit,
+    getFieldState,
+    handleChange,
+    arrayPush,
+    arrayUpdateByIndex,
+    arrayRemoveByIndex,
+  }: Form.api =
+    Form.use(
+      ~schema={
+        Form.Validation.Schema([|
+          Custom(
+            Name,
+            values =>
+              Js.String.length(values.name) < 5
+                ? Error("Invalid name") : Valid,
+          ),
+          Custom(
+            FavoriteColors,
+            ({favoriteColors}) => {
+              let length = Array.length(favoriteColors);
+
+              length < 0
+              || Belt.Array.some(favoriteColors, favColor =>
+                   Js.String.length(favColor.hex) == 0
+                 )
+                ? Error("Invalid colors") : Valid;
+            },
+          ),
+        |]);
+      },
+      ~onSubmit=({state}) => None,
+      ~initialState={name: "", favoriteColors: [||]},
+      (),
+    );
+
+  <form
+    onSubmit={event => {
+      ReactEvent.Synthetic.preventDefault(event);
+      submit();
+    }}>
+    <label>
+      <span> {"Name:" |> ReasonReact.string} </span>
+      <input
+        value={state.values.name}
+        onChange={ReForm.Helpers.handleDomFormChange(handleChange(Name))}
+      />
+      <p>
+        {getFieldState(Field(Name))
+         |> (
+           fun
+           | Error(error) => Some(error)
+           | _ => None
+         )
+         |> Belt.Option.getWithDefault(_, "")
+         |> ReasonReact.string}
+      </p>
+    </label>
+    <button onClick={_ => arrayPush(FavoriteColors, {name: "", hex: ""})}>
+      {React.string("Add color")}
+    </button>
+    {state.values.favoriteColors
+     ->Belt.Array.mapWithIndex((index, favColor) =>
+         <>
+           <hr />
+           <label>
+             <span> {"Name:" |> ReasonReact.string} </span>
+             <input
+               value={favColor.name}
+               onChange={ReForm.Helpers.handleDomFormChange(name =>
+                 arrayUpdateByIndex(
+                   ~field=FavoriteColors,
+                   ~index,
+                   {...favColor, name},
+                 )
+               )}
+             />
+           </label>
+           <label>
+             <span> {"Hex:" |> ReasonReact.string} </span>
+             <input
```

Comment:
with the nested schema do a way to have something like a `getArrayFieldState(~field: Config.field('a), ~index: int)`